### PR TITLE
Refactor: remove treelib; SELECT parser (SELECT+FROM+WHERE)

### PR DESCRIPTION
**Pull request by csongorszasz**
### Side note
This branch got tangled with another branch where I began working on the _SELECT_ command, but this is not a big deal. We shall let it go like this now, if you agree.

### About the pull request

As I was pondering about the implementation of _SELECT_, I realized that we don't need _treelib_. I completely removed every trace of the _treelib_ package because it didn't add to our project. In fact, it just created more headache, so I'm really sorry that I integrated it in the first place. 

I reran tests that previously passed and they still passed. This removal of _treelib_ should not have any impact on the app (it may even make it faster), and it opens up new possibilities as to how future functionalities could be implemented.

Feel free to pull the branch and test it for yourself, if you feel like it. Please tell me if you've got any concerns about this change.

### CODE REVIEW

File: server_side/interpreter/tree_objects/select.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+from server_side.interpreter.tree_objects.executable_tree import ExecutableTree
+
+
+class Select(ExecutableTree):
+    """
```
Comment (p3t9a) [inline server_side/interpreter/tree_objects/select.py:5]: Wow nice essay😁

**Review by p3t9a — COMMENTED** [review]
It looks good. I feel like you could keep working on select, and only merge this branch when it is done. If these change don't really make an impact on how everything works.
This way you could identify potential problems in the process (if there is any) and it will be more consistent. What do you think?

File: server_side/interpreter/tree_objects/select.py (hunk 1 above)
Comment (csongorszasz) [reply server_side/interpreter/tree_objects/select.py:5]: You're right. It's safer to merge this branch later.

Thank you, it's a good essay indeed :))

**Comment by csongorszasz** [comment]
SELECT parsing for:
- SELECT clause (todo: expressions inside select list)
- FROM clause
- WHERE clause

**Comment by csongorszasz** [comment]
Now the SELECT clause parsing can be said to be complete

**Review by p3t9a — APPROVED** [review]
Great work. Too much for me to read, but I see you tested it so it must be good.

**Pull request merged**